src/main.rs

```diff
@@ -4,44 +4,72 @@ use quicksilver::{
     Input, Result, Settings, Window,
 };
 
-use rand::distributions::{Distribution, Uniform};
+use rand::distributions::{Distribution, Uniform, WeightedIndex};
 use std::vec::Vec;
+use rand::prelude::ThreadRng;
 
+#[derive(Debug)]
 struct Node {
     x: f32,
     y: f32,
 }
 
+const HEIGHT: f32 = 1024.0;
+const LENGTH: f32 = 768.0;
+
+impl Node {
+    pub fn gen(mut rng: ThreadRng) -> Self {
```

Comment:
HEIGHT and LENGTH depend on a viewport of the application (and its not a window size, it's some fraction of the window).
For now, you can just pass HEIGHT and LENGTH as params.

Reply:
or make abstraction for viewport